### DISCUSSION TITLE
401 Unauthorized on deployed app

### DISCUSSION THREAD

**Discussion by Tobi-De** [fastapi/fastapi#3352] [Questions] [answered]
Hi, I have deployed a very simple fastapi app built using fastapi-users, tortoise-orm and postgres. I tested everything on my local machine and everything worked fine. I added a start event to create a verified superuser. I then deployed the app using docker and traefik on a vps, but all the routes that used the ``` current_user``` dependency all returned  a 401, not sure why. I thought this was an issue with the way the app was deployed, I removed traefik from the stack, removed the domain name, try a bunch of stuff, but still get 401 on the routes that were using  ```current_user```. I then deployed the app to heroku to see if that will change anything but I get the same 401 response, every other routes just worked fine. For the   ```current_user``` dependency I'm using ```current_user=fastapi_users.current_user(active=True, verified=False)```. Does anyone have any idea of ​​the problem? I'm probably doing something wrong somewhere but have no idea, any help is welcome, thanks.

**Comment by Tobi-De** [comment, marked as answer]
I already found the problem, this thread can be deleted

**Reply by ghandic** [reply]
@Tobi-De you can delete (or mark as answered) maybe just add a short comment of where you went wrong and mark your own comment as the answer to help others 😄

**Comment by Tobi-De** [comment]
> Hi, I have deployed a very simple fastapi app built using fastapi-users, tortoise-orm and postgres. I tested everything on my local machine and everything worked fine. I added a start event to create a verified superuser. I then deployed the app using docker and traefik on a vps, but all the routes that used the ` current_user` dependency all returned a 401, not sure why. I thought this was an issue with the way the app was deployed, I removed traefik from the stack, removed the domain name, try a bunch of stuff, but still get 401 on the routes that were using `current_user`. I then deployed the app to heroku to see if that will change anything but I get the same 401 response, every other routes just worked fine. For the `current_user` dependency I'm using `current_user=fastapi_users.current_user(active=True, verified=False)`. Does anyone have any idea of ​​the problem? I'm probably doing something wrong somewhere but have no idea, any help is welcome, thanks.

The problem was that my secret key was generated from a function and changed with every update of my server.

**Reply by ghandic** [reply]
That would do it 😁 thanks for adding the extra info to help others in the same situation 💫
